airbyte-cdk/python/airbyte_cdk/sources/declarative/retrievers/simple_retriever.py

```diff
@@ -406,13 +406,13 @@ def state(self, value: StreamState):
         """State setter, accept state serialized by state getter."""
         self.stream_slicer.update_cursor(value)
 
-    def _parse_records_and_emit_request_and_responses(self, request, response, stream_state, stream_slice) -> Iterable[StreamData]:
-        # Only emit requests and responses when running in debug mode
-        if self.logger.isEnabledFor(logging.DEBUG):
-            yield _prepared_request_to_airbyte_message(request)
-            yield _response_to_airbyte_message(response)
-        # Not great to need to call _read_pages which is a private method
-        # A better approach would be to extract the HTTP client from the HttpStream and call it directly from the HttpRequester
+    def parse_records(
```

Comment:
do we still need this method or can we call self.parse_response directly from https://github.com/airbytehq/airbyte/pull/24881/files#diff-574e7f982441af95b199fa0d628677b9a654543eaea1534b7516c8679375a235R371 and overwrite it in `SimpleRetrieverTestReadDecorator`?

Reply:
It isn't quite plug and play because part of what this intermediary `parse_records()` method is that it takes in request + response as parameters which we need in the decorated method to emit log messages. The original `parse_response()` method implementing the `HttpStream`'s abstract method doesn't have the necessary parameters so that's presumably why we had to add `parse_records()` on top